servicetalk-http-router-jersey-internal/src/main/java/io/servicetalk/http/router/jersey/internal/BufferPublisherInputStream.java

```diff
@@ -86,7 +86,8 @@ public int read(final byte[] b, final int off, final int len) throws IOException
     public void offloadSourcePublisher(final HttpExecutionStrategy executionStrategy,
                                 final Executor executor) {
         if (inputStream == EMPTY_INPUT_STREAM) {
-            publisher = executionStrategy.offloadReceive(executor, publisher);
+            publisher = executionStrategy.isMetadataReceiveOffloaded() || executionStrategy.isDataReceiveOffloaded() ?
```

Comment:
`executionStrategy.isMetadataReceiveOffloaded() ||` -> at this point we have already read the meta data and we are just focused on the payload, so I don't think we need to care about `isMetadataReceiveOffloaded`.

Reply:
Removing this causes ~16 test failures. I have retained it for now pending review of the test cases.

Reply:
Can you clarify which tests? If tests are verifying old behavior can they be updated, or do you think the tests are highlighting an issue where we shouldn't be using conditional offloading?

Reply:
Some examples:
```
Run JerseyRouterTestSuite with ServiceTalkJacksonSerializerAutoDiscoverable > JUnit Jupiter > ExecutionStrategyTest > testResource(TestExecutorStrategy, TestExecutorStrategy, TestExecutorStrategy, TestMode, String, RouterApi) > io.servicetalk.http.router.jersey.ExecutionStrategyTest.testResource(TestExecutorStrategy, TestExecutorStrategy, TestExecutorStrategy, TestMode, String, RouterApi)[166] FAILED
    java.lang.AssertionError: path=/rsc-default/subrsc-rte-exec-id-publisher-mapped, router=DEFAULT, class=DEFAULT, method=EXEC, mode=POST_RS : {"rs-thread":"servicetalk-global-executor-3-6","exec":"ContextPreservingStExecutor{delegate=DefaultExecutor{executor=InternalExecutor{service=ExecutorService@2f307179}, scheduler=SingleThreadedScheduler{offload=Executor@2f307179}}}","thread":"route-1458-1"}
    Expected: a string starting with "route-"
         but: was "servicetalk-global-executor-3-6"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.assertRouteExecutor(ExecutionStrategyTest.java:370)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.runTest(ExecutionStrategyTest.java:275)
        at io.servicetalk.http.router.jersey.AbstractJerseyStreamingHttpServiceTest.runTwiceToEnsureEndpointCache(AbstractJerseyStreamingHttpServiceTest.java:369)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.testResource(ExecutionStrategyTest.java:247)
```

```
Run JerseyRouterTestSuite with ServiceTalkJacksonSerializerAutoDiscoverable > JUnit Jupiter > ExecutionStrategyTest > testResource(TestExecutorStrategy, TestExecutorStrategy, TestExecutorStrategy, TestMode, String, RouterApi) > io.servicetalk.http.router.jersey.ExecutionStrategyTest.testResource(TestExecutorStrategy, TestExecutorStrategy, TestExecutorStrategy, TestMode, String, RouterApi)[491] FAILED
    java.lang.AssertionError: path=/rsc-default/subrsc-rte-exec-id-single-mapped, router=DEFAULT, class=DEFAULT, method=EXEC, mode=POST_RS : {"rs-thread":"stserverio-1440-2","exec":"ContextPreservingStExecutor{delegate=DefaultExecutor{executor=InternalExecutor{service=ExecutorService@2d8a8ba3}, scheduler=SingleThreadedScheduler{offload=Executor@2d8a8ba3}}}","thread":"route-2433-1"}
    Expected: a string starting with "route-"
         but: was "stserverio-1440-2"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.assertRouteExecutor(ExecutionStrategyTest.java:370)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.runTest(ExecutionStrategyTest.java:275)
        at io.servicetalk.http.router.jersey.AbstractJerseyStreamingHttpServiceTest.runTwiceToEnsureEndpointCache(AbstractJerseyStreamingHttpServiceTest.java:369)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.testResource(ExecutionStrategyTest.java:247)
```

In total it is 33 tests split between the Jersey and Jackson suites. The concern in this particular case is that removing `executionStrategy.isMetadataReceiveOffloaded() ||` was supposed to have no effect, but does appear to be changing something.

Reply:
sounds good. fine to leave as is 👍